src/devices/UFireIse/UFireIse.cs

```diff
@@ -0,0 +1,391 @@
+            _temperature = new Temperature(ReadRegister((byte)Register.ISE_TEMP_REGISTER), UnitsNet.Units.TemperatureUnit.DegreeCelsius);
+
+            return _temperature;
+        }
+
+        /// <summary>
+        /// Set temperature compensation
+        /// </summary>
+        /// <param name="temp">Compensation temperature</param>
+        public void SetTemperatureCompensation(Temperature temp)
+        {
+            WriteRegister((byte)Register.ISE_TEMP_REGISTER, Convert.ToSingle(temp.DegreesCelsius));
+            _temperature = temp;
+        }
+
+        /// <summary>
+        /// Calibrates the probe using a single point using a mV value.
+        /// </summary>
+        /// <param name="solution">Value. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateSingle(ElectricPotential solution)
+        {
+            if (solution.Millivolts >= -1024 && solution.Millivolts <= 1024)
+            {
+                WriteRegister((byte)Register.ISE_SOLUTION_REGISTER, Convert.ToSingle(solution.Millivolts));
+                SendCommand((byte)Command.ISE_CALIBRATE_SINGLE);
+
+                DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+            }
+        }
+
+        /// <summary>
+        /// Calibrates the dual-point values for the low reading, in mV, and saves them in the devices's EEPROM.
+        /// </summary>
+        /// <param name="solution">Value. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateProbeLow(ElectricPotential solution)
+        {
+            if (solution.Millivolts >= -1024 && solution.Millivolts <= 1024)
+            {
+                WriteRegister((byte)Register.ISE_SOLUTION_REGISTER, Convert.ToSingle(solution.Millivolts));
+                SendCommand((byte)Command.ISE_CALIBRATE_LOW);
+
+                DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+            }
+        }
+
+        /// <summary>
+        /// Calibrates the dual-point values for the high reading, in mV, and saves them in the devices's EEPROM.
+        /// </summary>
+        /// <param name="solution">Value. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateProbeHigh(ElectricPotential solution)
+        {
+            if (solution.Millivolts >= -1024 && solution.Millivolts <= 1024)
+            {
+                WriteRegister((byte)Register.ISE_SOLUTION_REGISTER, Convert.ToSingle(solution.Millivolts));
+                SendCommand((byte)Command.ISE_CALIBRATE_HIGH);
+
+                DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+            }
+        }
+
+        /// <summary>
+        /// Returns the firmware version of the device.
+        /// </summary>
+        /// <returns>Firmware version</returns>
+        public Version GetVersion()
+        {
+            return new Version(ReadRegister((byte)Register.ISE_VERSION_REGISTER).ToString());
```

Comment:
does it make sense to use constructor which doesn't take string? Is this major version? Minor? if so perhaps:
`return new Version(ReadRegister(....), 0)` or something along the lines